packages/react/src/themes/base/components/Icon/iconStyles.ts

```diff
@@ -0,0 +1,122 @@
+import * as _ from 'lodash'
+
+import icons from './index'
+import { callable, pxToRem, SizeValue } from '../../../../lib'
+import { ComponentSlotStylesInput, ICSSInJSStyle, FontIconSpec } from '../../../types'
+import { ResultOf } from '../../../../types'
+import { IconXSpacing, IconProps } from '../../../../components/Icon/Icon'
+import { IconVariables, IconSizeModifier } from './iconVariables'
+
+const sizes: Record<SizeValue, number> = {
+  smallest: 7,
+  smaller: 10,
+  small: 12,
+  medium: 16,
+  large: 20,
+  larger: 32,
+  largest: 40,
+}
+
+const getDefaultFontIcon = (iconName: string) => {
+  return callable(icons(iconName).icon)()
+}
+
+const getFontStyles = (
+  size: number,
+  iconName: string,
+  themeIcon?: ResultOf<FontIconSpec>,
+): ICSSInJSStyle => {
+  const { fontFamily, content } = themeIcon || getDefaultFontIcon(iconName)
+  const sizeInRems = pxToRem(size)
+
+  return {
+    fontFamily,
+    fontSize: sizeInRems,
+    lineHeight: 1,
+
+    display: 'inline-flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+
+    width: sizeInRems,
+    height: sizeInRems,
+
+    '::before': {
+      content,
+    },
+  }
+}
+
+const getXSpacingStyles = (xSpacing: IconXSpacing, horizontalSpace: string): ICSSInJSStyle => {
+  switch (xSpacing) {
+    case 'none':
+      return { marginLeft: 0, marginRight: 0 }
+    case 'before':
+      return { marginLeft: horizontalSpace, marginRight: 0 }
+    case 'after':
+      return { marginLeft: 0, marginRight: horizontalSpace }
+    case 'both':
+      return { marginLeft: horizontalSpace, marginRight: horizontalSpace }
+  }
+}
+
+const getBorderedStyles = (boxShadowColor: string): ICSSInJSStyle => {
+  return {
+    ...getPaddedStyle(),
+
+    boxShadow: `0 0 0 .05rem ${boxShadowColor} inset`,
+  }
+}
+
+const getPaddedStyle = (): ICSSInJSStyle => ({
+  padding: pxToRem(4),
+})
+
+const getIconSize = (size: SizeValue, sizeModifier: IconSizeModifier): number => {
+  if (!sizeModifier) {
+    return sizes[size]
+  }
+
+  const modifiedSizes = {
+    large: {
+      x: 24,
+      xx: 28,
```

Comment:
I'm not sure that this is part of base theme...

Reply:
Moved that back to the teams theme. The base theme now is always creating icon with fontSize of 16px